### DISCUSSION TITLE
API returned an empty page

### DISCUSSION THREAD

**Discussion by FeatherFatale** [victornpb/undiscord#290] [1. Questions & Answers]
Hello! For some reason, the script doesn't work. It was working fine for half of the messages but after that, it stopped working. And that happens with one person only. The script works for all my other chats. But with this person, it says total messages found:11459 but also says Ended because API returned an empty page. Can you help me?

**Comment by valkyrienyanko** [comment]
I believe this is because Discord does not remember the messages in a private DM that are older than some duration. Although it would be nice if this script did not rely on the API for this and attempted manual deletion.

**Comment by AstroOrbis** [comment]
I have this issue too ^^

**Comment by tacct6** [comment]
i had this issue on a public server after about 8000 messages/ messages older than ~6 months old too. it's worth noting before this happened, the deletion log slowly trickled down to fewer search results after each "page" (for example, something like: 25 results -> 20 -> 10 -> 5 -> 3 -> 1 -> 0). manually searching messages in the server still brought results up as expected so it seems like either a script issue or an anti-selfbot measure.

i swapped to a fork of this script with more recent updates (https://github.com/victornpb/deleteDiscordMessages/discussions/306) and was able to continue deleting messages in the same server without issue (+2000 so far) so until it's fixed on this repository i suggest trying the fork.

**Comment by aaa234243** [comment]
search for ur most recent message/within that timeframe (e.g. last message was on 1/1/2020 copy and paste the ID of one of the messages around then) and copy the ID of it or the persons message any ID around it and paste that into the "before message with ID section. "

**Comment by aijorgenson** [comment]
https://github.com/victornpb/deleteDiscordMessages/pull/323
